src/tests/SeleniumBrowser.php

```diff
@@ -0,0 +1,282 @@
+    function click( $element_or_selector, $parent = null, $timeout = null ) {
+        $element = $this->wait_for(
+            $element_or_selector,
+            $parent,
+            $timeout,
+            false,
+            // We need to use the full namespace here.
+            //See https://www.php.net/manual/en/language.types.callable.php#119166.
+            'Facebook\WebDriver\WebDriverExpectedCondition::elementToBeClickable'
+        );
+        $element->click();
+        return $element;
+    }
+
+    /**
+     * @param string $selector
+     * @param bool $multiple
+     * @return mixed|null
+     */
+    function get_element_if_it_exists( string $selector, bool $multiple=false ) {
+        try {
+            return $this->wait_for( $selector, null, 1, $multiple );
+        } catch ( Exception $ignore ) {
+            return null;
+        }
+    }
+
+    /**
+     * @param $input_or_selector
+     * @param $text
+     * @param $parent
+     * @return mixed
+     * @throws Exception
+     */
+    function fill_text_input( $input_or_selector, $text='', $parent=null ) {
+        $element = $this->click( $input_or_selector, $parent );
+        $this->click( $element );
+        $element->sendKeys( WebDriverKeys::CONTROL, 'a' );
+        $this->driver->wait( 0.1 );
+        $element->sendKeys( WebDriverKeys::DELETE );
+        if ( !empty( $text ) ) {
+            $element->sendKeys( $text );
+        }
+        return $element;
+    }
+
+    /**
+     * @return void
+     * @throws Exception
+     */
+    function log_into_wordpress() {
+        $this->open_page( WP_LOGIN_URL );
+        $this->fill_text_input( '#user_login', 'organic' );
+        $this->fill_text_input( '#user_pass', 'organic' );
+        $this->click( '#wp-submit' );
+        $upgrade_database = $this->get_element_if_it_exists( '[href^="upgrade.php"]' );
+        if ( !empty( $upgrade_database ) ) {
+            $this->click( $upgrade_database );
+        }
+        $this->wait( .5 );
+    }
+
+    /**
+     * @return void
+     * @throws Exception
+     */
+    function go_to_new_post() {
+        $this->open_page( WP_NEW_POST_URL );
+        $modal = $this->get_element_if_it_exists( '[aria-label="Close dialog"]' );
+        if ( !empty( $modal ) ) {
+            // Click out of the "getting started" modal if it exists.
+            $this->click( $modal );
+        }
+    }
+
+    /**
+     * @param string $block_type
+     * @return void
+     * Requires the browser to be in the editor.
+     * @throws Exception
+     */
+    function add_block( string $block_type ) {
+        // First, click the first Add block button.
+        $this->click( '[aria-label="Toggle block inserter"]' );
+        // Click the block search bar. Search for the block type.
+        $this->fill_text_input( '.components-search-control__input', $block_type );
+        // Click the icon for the block type to insert it.
+        $this->click( '.editor-block-list-item-' . $block_type );
+    }
```

Comment:
In general, version 5.9 seems flakier than version 6. There is an Add block button in the paragraph editor (first image below), but if I click that in 5.9, it sort of just disappears and then reappears? I've chosen to click the toolbar button (second image below) instead.

<img width="961" alt="Screen Shot 2023-04-14 at 5 34 24 PM" src="https://user-images.githubusercontent.com/47676832/232158969-94b64ec1-daa6-49f2-b2b1-5ebab67c404e.png">

<img width="1448" alt="Screen Shot 2023-04-14 at 5 34 28 PM" src="https://user-images.githubusercontent.com/47676832/232158986-0c927c1a-9e4d-4fbf-a040-073ef5099b88.png">